### PR TITLE
Add stress test benchmarking script

### PR DESCRIPTION
**Pull request by msimberg**
This adds the "stress test" script that I used for the previous benchmarking session to check how reliably we can run the eigensolvers.

It generates n jobs, where each job will run the eigensolver repeatedly with a single matrix size and block size on a random number of nodes. All results are checked. The seed used for generating the random sizes is stored so that the jobs can be recreated if needed. All runs have their results checked.

My python knowledge for handling the seed may be somewhat suboptimal. I'm currently storing the seed as an environment variable in the job files. I'm also converting the seed from a string of unix seconds to an int which feels wrong. Suggestions for improvements welcome.

### CODE REVIEW

File: scripts/gen_dlaf_random-gpu.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    mb_szs = min(m_szs, mb_szs)
+
+    full_kwargs = kwargs.copy()
+    full_kwargs["env"] = f'DLAF_STRESS_TEST_SEED="{seed}"'
+    full_kwargs["lib"] = "dlaf"
+    full_kwargs["miniapp_dir"] = dlafpath
+    full_kwargs["nruns"] = nruns
+    full_kwargs["dtype"] = dtype
+
+    run = mp.StrongScaling(system, "DLAF_test_strong", "job_dlaf", nodes_arr, time)
```
[the remark this answers is not in the document]
Comment (msimberg) [reply scripts/gen_dlaf_random-gpu.py]: I made it `DLAF_test_stress`, just to keep with the naming scheme of the strong/weak scaling scripts.

File: scripts/gen_dlaf_random-gpu.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+njobs = 100
+
+nodes_arr = [0.25, 0.5, 1, 2, 4, 8, 16, 32]
+rpn = 4
+m_szs_d = (1000, 50000)
+mb_szs_d = (200, 4000)
+m_szs_z = (1000, 25000)
+mb_szs_z = (200, 4000)
+
+extra_flags = "--pika:threads=64 --nwarmups=0 --check=all"
```
[the remark this answers is not in the document]
Comment (msimberg) [reply scripts/gen_dlaf_random-gpu.py:39]: Yeah, might be a good idea. That's probably worth discussing in today's meeting.

[the remark this answers is not in the document]
Comment (rasolca) [reply scripts/gen_dlaf_random-gpu.py:39]: Especially for miniapp scripts I prefer to have everything in the script.
It is much easier to keep the full script without command line args for reproducibility reasons.

[the remark this answers is not in the document]
Comment (RMeli) [reply scripts/gen_dlaf_random-gpu.py:39]: If you use the defaults in the way you suggest, you still have reproducibility, while also allowing for flexibility when trying things out.

**Review by rasolca — APPROVED** [review]

**Pull request merged**